src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2665,6 +2677,10 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
 
             _snapshotBuilderOpt?.ExitWalker(this.SaveSharedState(), previousSlot);
 
+            _variableTypes.Free();
+            _variableTypes = oldVariableTypes;
+            _variableSlot.Free();
+            _variableSlot = oldVariableSlot;
```

Comment:
>_variableSlot [](start = 12, length = 13)

Is it possible that, while analyzing a local function, we allocate a slot for something that is not scoped to the body of the function and, therefore, the slot should be shared with containing method?

Reply:
The containing method state is reset after the nested function is visited so any slots added in the method can be ignored.

---
In reply to: [535419178](https://github.com/dotnet/roslyn/pull/49765#discussion_r535419178) [](ancestors = 535419178)

Reply:
> The containing method state is reset after the nested function is visited so any slots added in the method can be ignored.

That may not be true. `_localFuncVarUsages` may have been updated and that tracks state as well.

---
In reply to: [535547721](https://github.com/dotnet/roslyn/pull/49765#discussion_r535547721) [](ancestors = 535547721,535419178)